.github/scripts/pr_body_check.py

```
@@ -0,0 +1,68 @@
+    # Match lines like: ### 🧑‍🔬 Unit Testing
+    section_regex = re.compile(
+        r'^\s*#{2,6}\s*([\W_]{1,4} [^\n]+)\n([\s\S]*?)(?=^\s*#{2,6}\s*[\W_]{1,4} [^\n]+\n|$)',
+        re.MULTILINE,
+    )
+    return {m.group(1).strip(): m.group(2).strip() for m in section_regex.finditer(text)}
+
+def has_checked_box(text: str) -> bool:
+    # Looks for - [x] or - [X] at the start of a line
+    return bool(re.search(r'^\s*-\s*\[[xX]\]', text, re.MULTILINE))
+
+def main() -> None:
```

Ruff (0.12.2)

`main` is too complex (11 > 10)

(C901)